sound/soc/sof/intel/hda.c

```diff
@@ -581,6 +580,18 @@ static const char *fixup_tplg_name(struct snd_sof_dev *sdev,
 
 #endif
 
+static bool hda_use_common_hdmi(void)
+{
+#if !IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI) && \
+	IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)
+
+	if (hda_codec_use_common_hdmi)
+		return false;
```

Comment:
let me try to understand: if Kconfig is right for this code to enable, it would mean: if hda_codec_use_common_hdmi is true, we return false. If it's false, we fall through and return its value below, which is false. So, we return false in both cases. I think it would be much clearer to write this
```
#if !IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI) && \
	IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)
	return false;
#else
	return hda_codec_use_common_hdmi;
#endif
```
What if none of those CONFIG_* options is enabled? Don't we have to return false too? Even if it doesn't make any difference, that seems a more logical definition to me - if `CONFIG_SND_HDA_CODEC_HDMI` isn't enabled, return false. So, maybe you actually can just do `return IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI) && hda_codec_use_common_hdmi;`

Reply:
@lyakh Ack, agreed, returning false right away is better. 
I do want to return true in the case neither is selected. The common hda driver path does an explicit driver probe&attach and we can report errors immediately in hda_codec_probe(). And more importantly, we can modify the codec_mask and disable HDMI and continue SOF load. With hdac-hdmi, we find the missing codec much later and then we just fail in topology load and user is left without sound. So basicly this should return false only and only if user really, really wants to use hdac-hdmi.

Reply:
@kv2019i mach_params->common_hdmi_codec_drv i set to hda_codec_use_common_hdmi, should this be set to hda_use_common_hdmi() instead?

Reply:
@ranj063 Thanks, good catch. I had overlooked that, needs to be changed as well.